src/Api/Controllers/GroupsController.cs

```diff
@@ -38,13 +41,20 @@ public class GroupsController : Controller
         _createGroupCommand = createGroupCommand;
         _updateGroupCommand = updateGroupCommand;
         _deleteGroupCommand = deleteGroupCommand;
+        _authorizationService = authorizationService;
     }
 
     [HttpGet("{id}")]
     public async Task<GroupResponseModel> Get(string orgId, string id)
     {
         var group = await _groupRepository.GetByIdAsync(new Guid(id));
-        if (group == null || !await _currentContext.ManageGroups(group.OrganizationId))
+        if (group == null)
+        {
+            throw new NotFoundException();
+        }
+
+        var authorizationResult = await _authorizationService.AuthorizeAsync(User, group, GroupOperations.Read);
```

Comment:
If your authorization fails on null group, you can avoid the null checks above.

I think this is reasonable behavior -- can I read this group? No, that group doesn't exist

Reply:
I forgot to push these changes yesterday, but I also moved away from `NotFoundException` for unauthorized access. If we don't want to distinguish between "not found" and "not authorized" then I agree, but I thought it better to return accurate error codes.

Reply:
In that case, just throw inside the Authorization method 🤷

Reply:
We want to maintain the `NotFoundException` for both cases, so I've moved it into the authHandler as you suggest.